internal/telemetry/wrapper.go

```diff
@@ -0,0 +1,36 @@
+package telemetry
+
+import (
+	"context"
+	"github.com/pterm/pterm"
+)
+
+// Wrapper wraps the function calls with the telemetry handlers
+func Wrapper(ctx context.Context, et EventType, f func() error) (err error) {
+	cli := Get()
+
+	attemptSuccessFailure := true
+
+	if err := cli.Start(ctx, et); err != nil {
+		pterm.Debug.Printfln("Unable to send telemetry start data: %s", err)
+		attemptSuccessFailure = false
+	}
+
+	defer func() {
```

Comment:
I think the usage of a `defer` here to handle the errors/success reporting makes this harder to read and introduces the need for using the named return. 

It would be cleaner to just  call `f()`, capture the error and check if it was `nil` or not. If I'm missing some context about why you prefer to use the defer let me know.